$OpenBSD: patch-amavisd,v 1.13 2016/03/02 14:30:50 sthen Exp $

Hunks 1, 3: Disable File::LibMagic in favour of safer file(1) from base.

--- amavisd.orig	Sun Oct 26 00:06:00 2014
+++ amavisd	Tue Feb  2 22:08:33 2016
@@ -12557,7 +12557,7 @@ sub after_chroot_init() {
                grep(/\.pm\z/, keys %INC)) {
     next  if !grep($_ eq $m, qw(Amavis::Conf
       Archive::Tar Archive::Zip Compress::Zlib Compress::Raw::Zlib
-      Convert::TNEF Convert::UUlib File::LibMagic
+      Convert::TNEF Convert::UUlib
       MIME::Entity MIME::Parser MIME::Tools Mail::Header Mail::Internet
       Digest::MD5 Digest::SHA Digest::SHA1 Crypt::OpenSSL::RSA
       Authen::SASL Authen::SASL::XS Authen::SASL::Cyrus Authen::SASL::Perl
@@ -29847,7 +29847,7 @@ sub new_SpamAssassin_instance {
 #   PREFIX            => '/usr/local',
 #   DEF_RULES_DIR     => '/usr/local/share/spamassassin',
 #   LOCAL_RULES_DIR   => '/etc/mail/spamassassin',
-#   LOCAL_STATE_DIR   => '/var/lib/spamassassin',
+    LOCAL_STATE_DIR   => '/var/db/spamassassin',
 #see Mail::SpamAssassin man page for other options
   };
   if ($sa_version_num < 3.001005 && !defined $sa_args->{LOCAL_STATE_DIR})
@@ -30595,17 +30595,8 @@ BEGIN {
   import Amavis::Unpackers::NewFilename qw(consumed_bytes);
 }
 
-BEGIN {
   use vars qw($filemagic);
-  eval {
-    require File::LibMagic;
-    File::LibMagic->VERSION(1.00);
-    import File::LibMagic;
-    $filemagic = File::LibMagic->new;
-  } or do {
     undef $filemagic;
-  };
-}
 
 use subs @EXPORT_OK;
 
